core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -145,17 +149,22 @@ private LocalDate getCurrentDayDate(UserAnonymizedDTO userAnonymized)
 
 	private <T extends IntervalActivity> void addMissingInactivity(Goal activeGoal, ZonedDateTime dateAtStartOfInterval,
 			Set<T> activityEntitiesAtDate, UserAnonymizedDTO userAnonymized,
-			BiFunction<Goal, ZonedDateTime, T> inactivityEntitySupplier)
+			BiFunction<Goal, ZonedDateTime, T> inactivityEntitySupplier, Consumer<T> existingEntityInactivityCompletor)
 	{
-		if (!containsActivityForGoal(activityEntitiesAtDate, activeGoal))
+		Optional<T> activityForGoal = getActivityForGoal(activityEntitiesAtDate, activeGoal);
+		if (activityForGoal.isPresent())
+		{
+			existingEntityInactivityCompletor.accept(activityForGoal.get());
```

Comment:
Added a comment to explain. Even if activity was already recorded (e.g. a `WeekActivity`), it might be that this is not for the complete period (e.g. only for one day in that week), so make the interval activity complete (e.g. add `DayActivity` inactivity) with a consumer.